slo-workload/src/Commands/RunCommand.php

```diff
@@ -0,0 +1,370 @@
+            CURLOPT_RETURNTRANSFER => true,
+            CURLOPT_ENCODING => '',
+            CURLOPT_MAXREDIRS => 10,
+            CURLOPT_TIMEOUT => 0,
+            CURLOPT_FOLLOWLOCATION => true,
+            CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
+            CURLOPT_CUSTOMREQUEST => 'DELETE',
+        ));
+
+        $response = curl_exec($curl);
+
+        curl_close($curl);
+    }
+
+    function forkJob(Closure $function, int $count): array
+    {
+        $pIds = [];
+        for ($i = 0; $i < $count; $i++) {
+            $pid = pcntl_fork();
+            if ($pid == -1) {
+                echo "Error fork";
+                exit(1);
+            } elseif ($pid == 0) {
+                try {
+                    $function($i);
+                } catch (\Exception $e) {
+                    echo "Error on $i'th fork: " . $e->getMessage();
+                }
+                exit(0);
+            } else {
+                $pIds[] = $pid;
+                usleep($i * 1e3);
+            }
+        }
+        return $pIds;
+    }
+
+    protected function readJob(string $endpoint, string $path, $tableName, int $initialDataCount, int $time, int $readTimeout, int $process, int $shutdownTime, $startTime)
+    {
+        $ydb = Utils::initDriver($endpoint, $path, "read-$process");
+        $dataGenerator = new DataGenerator($initialDataCount);
+        $query = sprintf(Defaults::READ_QUERY, $tableName);
+        $table = $ydb->table();
+
+        while (microtime(true) <= $startTime + $time) {
+            $begin = microtime(true);
+            Utils::metricsStart("read", $this->queueId);
+            $attemps = 0;
+            $id = (new Uint64Type($dataGenerator->getRandomId()))->toTypedValue();
+            try {
+                $table->retryTransaction(function (Session $session)
+                use ($id, $query, $dataGenerator, $tableName, &$attemps) {
+                    try {
+                        $session->query($query, [
+                            "\$id" => $id
+                        ]);
+                        $attemps++;
+                    } catch (\Exception $exception) {
+                        Utils::retriedError($this->queueId, 'read', get_class($exception));
+                    }
+                }, true, null, [
+                    'callback_on_error' => function (\Exception $e) use (&$attemps) {
+                        $attemps++;
+                        Utils::retriedError($this->queueId, 'write', get_class($e));
+                    }
+                ]);
+                Utils::metricDone("read", $this->queueId, $attemps, $this->getLatency($begin));
+            } catch (\Exception $e) {
+                $table->getLogger()->error($e->getMessage());
+                Utils::metricFail("read", $this->queueId, $attemps, get_class($e), $this->getLatency($begin));
+            } finally {
+                $delay = $begin * 1e6 + 1e6 / Defaults::RPS_PER_WRITE_FORK - microtime(true) * 1e6;
```

Comment:
the code used twice.

What about extract function?:
calc_delay($start_mictotime, $begin_last_query_microtime, $rps).

$start_mictotime - time of first query
$begin_last_query_microtime - begin of last query

it will allow get better precision for rps